contracts/proxy-lib/src/lib.rs

```diff
@@ -0,0 +1,408 @@
+            ********************************/
+            self.execute_proposal(proposal);
+        } else {
+            approvals.insert(signer_id);
+        }
+    }
+
+    fn perform_action_by_member(&self, action: MemberAction) -> Promise {
+        let identity = match &action {
+            MemberAction::Approve { identity, .. } => identity,
+            MemberAction::Create { proposal, .. } => &proposal.author_id,
+        }
+        .rt()
+        .expect("Could not transmute");
+        config_contract::ext(self.context_config_account_id.clone())
+            .has_member(Repr::new(self.context_id), identity)
+            .then(match action {
+                MemberAction::Approve {
+                    identity,
+                    proposal_id,
+                } => Self::ext(env::current_account_id())
+                    .internal_approve_proposal(identity, proposal_id),
+                MemberAction::Create {
+                    proposal,
+                    num_proposals,
+                } => Self::ext(env::current_account_id())
+                    .internal_create_proposal(proposal, num_proposals),
+            })
+    }
+
+    pub fn proposals(&self, offset: usize, length: usize) -> Vec<(&u32, &Proposal)> {
+        let effective_len = (self.proposals.len() as usize)
+            .saturating_sub(offset)
+            .min(length);
+        let mut proposals = Vec::with_capacity(effective_len);
+        for proposal in self.proposals.iter().skip(offset).take(length) {
+            proposals.push(proposal);
+        }
+        proposals
+    }
+
+    pub fn get_confirmations_count(
+        &self,
+        proposal_id: ProposalId,
+    ) -> Option<ProposalWithApprovals> {
+        let approvals_for_proposal = self.approvals.get(&proposal_id);
+        match approvals_for_proposal {
+            Some(approvals) => Some(ProposalWithApprovals {
+                proposal_id,
+                num_approvals: approvals.len(),
+            }),
+            None => None,
+        }
+    }
+
+    #[private]
+    pub fn internal_approve_proposal(
+        &mut self,
+        signer_id: Repr<SignerId>,
+        proposal_id: ProposalId,
+        #[callback_result] call_result: Result<bool, PromiseError>, // Match the return type
+    ) -> Option<ProposalWithApprovals> {
+        assert_membership(call_result);
+
+        self.internal_confirm(proposal_id, signer_id.rt().expect("Invalid signer"));
+        self.build_proposal_response(proposal_id)
+    }
+
+    #[private]
+    pub fn internal_create_proposal(
+        &mut self,
+        proposal: Proposal,
+        num_proposals: u32,
+        #[callback_result] call_result: Result<bool, PromiseError>, // Match the return type
+    ) -> Option<ProposalWithApprovals> {
+        assert_membership(call_result);
+
+        self.num_proposals_pk
+            .insert(*proposal.author_id, num_proposals);
+
+        let proposal_id = self.proposal_nonce;
```

Comment:
to match SDK expectations, proposal ID should be random and determined by the runtime, but this can come in a separate PR, just noting it

Reply:
@danwilliams plese change it once you are generating the Proposal Id